pkg/dns/aws/dns.go

```diff
@@ -198,9 +198,11 @@ func (m *Provider) getLBHostedZone(name string) (string, error) {
 	var id string
 	fn := func(resp *elb.DescribeLoadBalancersOutput, lastPage bool) (shouldContinue bool) {
 		for _, lb := range resp.LoadBalancerDescriptions {
-			log.V(2).Info("found load balancer", "name", aws.StringValue(lb.LoadBalancerName), "dns name", aws.StringValue(lb.DNSName), "hosted zone ID", aws.StringValue(lb.CanonicalHostedZoneNameID))
-			if aws.StringValue(lb.CanonicalHostedZoneName) == name {
-				id = aws.StringValue(lb.CanonicalHostedZoneNameID)
+			dnsName := aws.StringValue(lb.DNSName)
+			zoneID := aws.StringValue(lb.CanonicalHostedZoneNameID)
+			log.V(2).Info("found load balancer", "name", aws.StringValue(lb.LoadBalancerName), "dns name", dnsName, "hosted zone ID", zoneID)
+			if dnsName == name {
```

Comment:
What was the motivation for this change?

Reply:
Oh, because AWS doesn't set `CanonicalHostedZoneName` on internal LBs.  Never mind.

Reply:
I put this change in its own commit (5eee64ddcc678b552e8a4e59145ac73ec89e9752) which explains it

Reply:
Yeah, the cloud provider code uses `lb.DNSName` for the name that ultimately goes into the `Service` object's `status` ([src](https://github.com/kubernetes/cloud-provider-aws/blob/87ce39a8e225fb0a76cb717de63bedc64cc0af89/pkg/cloudprovider/providers/aws/aws.go#L3675)), so we should be using the same to relate the `Service` back to the LB.

Reply:
Something I'd like to discuss upstream is, if the DNS name alone is insufficient to identify the LB (i.e. it must be combined with the hosted zone ID), the cloud provider could perhaps surface the zone ID so that we don't need to do any lookups at all (which is pretty brittle/unsafe)